### PR TITLE
python310Packages.numpyro: add missing input for tests

### PR DESCRIPTION
**Pull request by fabaff**
###### Description of changes
Fix build (https://hydra.nixos.org/build/219639457)

ZHF: #230712

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 231808` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>48 packages failed to build:</summary>
  <ul>
    <li>python310Packages.arviz</li>
    <li>python310Packages.arviz.dist</li>
    <li>python310Packages.bambi</li>
    <li>python310Packages.bambi.dist</li>
    <li>python310Packages.blackjax</li>
    <li>python310Packages.blackjax.dist</li>
    <li>python310Packages.h5netcdf</li>
    <li>python310Packages.h5netcdf.dist</li>
    <li>python310Packages.jaxopt</li>
    <li>python310Packages.jaxopt.dist</li>
    <li>python310Packages.pymc</li>
    <li>python310Packages.pymc.dist</li>
    <li>python310Packages.qiskit</li>
    <li>python310Packages.qiskit-aer</li>
    <li>python310Packages.qiskit-aer.dist</li>
    <li>python310Packages.qiskit-finance</li>
    <li>python310Packages.qiskit-finance.dist</li>
    <li>python310Packages.qiskit-ibmq-provider</li>
    <li>python310Packages.qiskit-ibmq-provider.dist</li>
    <li>python310Packages.qiskit-ignis</li>
    <li>python310Packages.qiskit-ignis.dist</li>
    <li>python310Packages.qiskit-nature</li>
    <li>python310Packages.qiskit-nature.dist</li>
    <li>python310Packages.qiskit-optimization</li>
    <li>python310Packages.qiskit-optimization.dist</li>
    <li>python310Packages.qiskit.dist</li>
    <li>python310Packages.sunpy</li>
    <li>python310Packages.sunpy.dist</li>
    <li>python311Packages.blackjax</li>
    <li>python311Packages.blackjax.dist</li>
    <li>python311Packages.h5netcdf</li>
    <li>python311Packages.h5netcdf.dist</li>
    <li>python311Packages.jaxopt</li>
    <li>python311Packages.jaxopt.dist</li>
    <li>python311Packages.qiskit-aer</li>
    <li>python311Packages.qiskit-aer.dist</li>
    <li>python311Packages.qiskit-finance</li>
    <li>python311Packages.qiskit-finance.dist</li>
    <li>python311Packages.qiskit-ibmq-provider</li>
    <li>python311Packages.qiskit-ibmq-provider.dist</li>
    <li>python311Packages.qiskit-ignis</li>
    <li>python311Packages.qiskit-ignis.dist</li>
    <li>python311Packages.qiskit-nature</li>
    <li>python311Packages.qiskit-nature.dist</li>
    <li>python311Packages.qiskit-optimization</li>
    <li>python311Packages.qiskit-optimization.dist</li>
    <li>python311Packages.sunpy</li>
    <li>python311Packages.sunpy.dist</li>
  </ul>
</details>
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>python310Packages.cvxpy</li>
    <li>python310Packages.cvxpy.dist</li>
    <li>python310Packages.numpyro</li>
    <li>python310Packages.numpyro.dist</li>
    <li>python310Packages.qutip</li>
    <li>python310Packages.qutip.dist</li>
    <li>python311Packages.cvxpy</li>
    <li>python311Packages.cvxpy.dist</li>
    <li>python311Packages.qutip</li>
    <li>python311Packages.qutip.dist</li>
  </ul>
</details>

**Comment by risicle** [comment]
Does this.. _depend_ on #231842?

File: pkgs/development/python-modules/h5netcdf/default.nix [hunk 1]
```diff
@@ -21,9 +21,11 @@ buildPythonPackage rec {
...
   };
 
- nativeBuildInputs = [
-   setuptools-scm
- ];
+  SETUPTOOLS_SCM_PRETEND_VERSION = version;
+
+   nativeBuildInputs = [
+     setuptools-scm
+   ];
```
[the remark this answers is not in the document]
Comment (fabaff) [reply pkgs/development/python-modules/h5netcdf/default.nix]: Fixed, thanks.

File: pkgs/development/python-modules/h5netcdf/default.nix [hunk 2]
```diff
@@ -10,7 +10,7 @@
 }:
 
 buildPythonPackage rec {
-  pname = "h5netcdf";
+  pname = "git comm";
```
[the remark this answers is not in the document]
Comment (fabaff) [reply pkgs/development/python-modules/h5netcdf/default.nix]: Fixed, thanks.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 231808` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages failed to build:</summary>
  <ul>
    <li>python310Packages.bambi</li>
    <li>python310Packages.bambi.dist</li>
    <li>python310Packages.pymc</li>
    <li>python310Packages.pymc.dist</li>
    <li>python310Packages.sunpy</li>
    <li>python310Packages.sunpy.dist</li>
    <li>python311Packages.sunpy</li>
    <li>python311Packages.sunpy.dist</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python310Packages.arviz</li>
    <li>python310Packages.arviz.dist</li>
    <li>python310Packages.h5netcdf</li>
    <li>python310Packages.h5netcdf.dist</li>
    <li>python310Packages.numpyro</li>
    <li>python310Packages.numpyro.dist</li>
    <li>python311Packages.h5netcdf</li>
    <li>python311Packages.h5netcdf.dist</li>
  </ul>
</details>

**Comment by fabaff** [comment]
`sunpy`, `pymc` and `bambi` are already broken.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 231808` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages failed to build:</summary>
  <ul>
    <li>python310Packages.bambi</li>
    <li>python310Packages.bambi.dist</li>
    <li>python310Packages.pymc</li>
    <li>python310Packages.pymc.dist</li>
    <li>python310Packages.sunpy</li>
    <li>python310Packages.sunpy.dist</li>
    <li>python311Packages.sunpy</li>
    <li>python311Packages.sunpy.dist</li>
  </ul>
</details>
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python310Packages.arviz</li>
    <li>python310Packages.arviz.dist</li>
    <li>python310Packages.h5netcdf</li>
    <li>python310Packages.h5netcdf.dist</li>
    <li>python310Packages.numpyro</li>
    <li>python310Packages.numpyro.dist</li>
    <li>python311Packages.h5netcdf</li>
    <li>python311Packages.h5netcdf.dist</li>
  </ul>
</details>

**Pull request merged**